pkg/inject/inject_test.go

```diff
@@ -288,3 +291,27 @@ func TestGetOverriddenValues(t *testing.T) {
 		})
 	}
 }
+
+func TestWholeCPUCores(t *testing.T) {
+	for _, c := range []struct {
+		v string
+		n int
+	}{
+		{v: "1", n: 1},
+		{v: "1m", n: 1},
+		{v: "1000m", n: 1},
+		{v: "1001m", n: 2},
```

Comment:
nit: why not round instead? so that this is `1` and not `2` so that we are not going out of the boundary :thinking:

Reply:
(1) quantity exposes a `RoundUp` and not `Round`
(2) i think if you say the proxy can use more than a core and we only allocate a single core, we'll never use the full limit, so rounding up better-reflects the user's intent, I think